src/backend/FramebufferBackend.cc

```diff
@@ -0,0 +1,200 @@
+    struct fb_fix_screeninfo finfo;
+
+    // Get the screen info.
+    if (ioctl(this->fbfd, FBIOGET_VSCREENINFO, &vinfo)) {
+        this->destroySurface();
+        Nan::ThrowError("Error retrieving data from framebuffer");
+        return nullptr;
+    }
+
+    // Store the original screen info for restoration later.
+    memcpy(&this->origVinfo, &vinfo, sizeof(struct fb_var_screeninfo));
+
+    // Setup the number of bits per pixel.
+    vinfo.bits_per_pixel = 8;
+    if (ioctl(this->fbfd, FBIOPUT_VSCREENINFO, &vinfo)) {
+        this->destroySurface();
+        Nan::ThrowError("Error sending data to framebuffer");
+        return nullptr;
+    }
+
+    // Get the length of the screen buffer in bytes.
+    if (ioctl(this->fbfd, FBIOGET_FSCREENINFO, &finfo)) {
+        this->destroySurface();
+        Nan::ThrowError("Error retrieving data from framebuffer");
+        return nullptr;
+    }
+    this->screenSize = finfo.smem_len;
+
+    // Memory map the framebuffer.
+    this->fbp = (char*)mmap(0,
+                            this->screenSize,
+                            PROT_READ | PROT_WRITE,
+                            MAP_SHARED,
+                            this->fbfd,
+                            0);
+
+    if (this->fbp == MAP_FAILED) {
+        this->destroySurface();
+        Nan::ThrowError("Error during memory mapping");
+        return nullptr;
+    }
+
+    // Create a cairo surface from the memory mapped buffer.
+    screenSurface = cairo_image_surface_create_for_data(
+            (unsigned char *)this->fbp,
+            CAIRO_FORMAT_RGB16_565,
+            vinfo.xres, vinfo.yres,
+            cairo_format_stride_for_width(CAIRO_FORMAT_RGB16_565, vinfo.xres));
+
+    // Store the width and height to be used further up the stack.
+    this->width = vinfo.xres;
+    this->height = vinfo.yres;
+
+    // If double buffering is enabled then allocate an in memory buffer.
+    if (this->doubleBufferEnabled) {
+        this->bbp = (char *)malloc(this->screenSize);
+        bufferSurface = cairo_image_surface_create_for_data(
+                (unsigned char *)this->bbp,
+                CAIRO_FORMAT_RGB16_565,
+                vinfo.xres, vinfo.yres,
+                cairo_format_stride_for_width(CAIRO_FORMAT_RGB16_565, vinfo.xres));
+
+        if (cairo_surface_status(bufferSurface) != CAIRO_STATUS_SUCCESS) {
+            Nan::ThrowError("Error creating buffer surface");
+            this->destroySurface();
+            return nullptr;
+        }
+        this->surface = bufferSurface;
+        return bufferSurface;
+    } else {
+        this->surface = screenSurface;
+        return screenSurface;
+    }
+}
+
+void FramebufferBackend::setWidth(int width)
+{
+    Nan::ThrowError("Framebuffer setWidth is not supported");
+}
+
+void FramebufferBackend::setHeight(int height)
+{
+    Nan::ThrowError("Framebuffer setHeight is not supported");
+}
+
+cairo_format_t FramebufferBackend::getFormat()
+{
+    return CAIRO_FORMAT_RGB16_565;
```

Comment:
Why is `CAIRO_FORMAT_RGB16_565` format hardcoded? Receng graphic cards use RGBA32 by default... Currently RGB16 is only being used by default QEmu Cirrus VGA graphic card, and it's heavily recomended to use the SVGA one that between other things don't have this limitation.

Reply:
Tasty Pi's :) You're right though, this should really be a parameter

Reply:
If you take a look on my pull-request, I have support for several color schemes there :-)